packages/module/src/ExpandableText/ExpandableText.tsx

```diff
@@ -0,0 +1,99 @@
+
+const useStyles = createUseStyles({
+  expandableText: {
+    "& button": {
+      padding: "0"
+    },
+    "& is-inline": { marginRight: [ "10px", "0.625rem" ] }
+  } });
+
+const dangerousHtml = (html: string) => ({ __html: sanitizeHtml(html) });
+
+export interface ExpandableTextProps {
+  /** Additional classes added to the ExpandableText */
+  className?: string;
+  /** Text to display in the ExpandableText component */
+  text?: string;
+  /** A set length for the ExpandableText component. */
+  length?: number;
+  /** Indicates if the component is expanded to show the entire text */
+  expandText?: React.ReactNode;
+  /** Indicates if the component is collapsed to show the truncated text */
+  collapseText?: React.ReactNode;
+  /** Indicates if the text should be kept inline */
+  inline?: boolean;
+  spaceBetween?: boolean;
+  hideExpandText?: boolean;
+  /** If set to true the text will be expanded when moused over */
+  expandOnMouseOver?: boolean;
+}
+
+const ExpandableText: React.FunctionComponent<ExpandableTextProps> = ({
+  text = '',
+  length = 150,
+  expandText = 'Read more',
+  hideExpandText = false,
+  expandOnMouseOver = false,
+  collapseText = 'Collapse',
+  className,
+  inline,
+  spaceBetween,
+  ...props
+}: ExpandableTextProps) => {
+  const classes = useStyles();
+  const expandableTextClasses = classNames(classes.expandableText, className, { [`is-inline`]: inline }, { [`is-block`]: !inline });
+  const trimmedText = text.substring(0, length);
+  const textOverflow = text.length > length;
+  const [ showText, setShowText ] = useState(false);
+  const toggleText = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event && event.preventDefault();
+    setShowText(!showText);
+  };
+
+  const expandButton = (
+    <Button variant="link" onClick={toggleText}>
+      {expandText}
+    </Button>
+  );
+  const collapseButton = (
+    <Button variant="link" onClick={toggleText}>
+      {collapseText}
+    </Button>
+  );
+  const textWithOverflow = showText === false ? `${trimmedText}${textOverflow ? '...' : ''}` : text;
+  const html = dangerousHtml(textWithOverflow);
+  const mouseOverHandler = expandOnMouseOver && {
+    onMouseEnter: () => setShowText(true),
+    onMouseLeave: () => setShowText(false),
+  };
+
+  return inline ? (
+    <React.Fragment>
+      <span className={expandableTextClasses} widget-type="PFComponentGroupExpandableTextInline" dangerouslySetInnerHTML={html} {...mouseOverHandler} {...props} />
+      {!hideExpandText && textOverflow && (showText === false ? expandButton : collapseButton)}
+    </React.Fragment>
+  ) : (
+    <Stack className={expandableTextClasses} {...props}>
+      <StackItem {...mouseOverHandler}>
+        <span widget-type="PFComponentGroupExpandableTextBlock" dangerouslySetInnerHTML={html} />
+      </StackItem>
+      {!hideExpandText && textOverflow && (
+        <StackItem
+          className={classNames({
+            'pf-u-mt-sm': spaceBetween,
```

Comment:
```suggestion
            'pf-v5-u-mt-sm': spaceBetween,
```
without the v5 the className does not work

but thinking if it's not better to just enable user to pass a completely custom button or replace the `spaceBetween` with normal ButtonProps omitting the `onClick`. What do you think?

Reply:
I think for now we can add the v5.  If we get get a request for a custom button then we can add that.

Reply:
Updated based on our discussion yesterday to use the custom button as we discussed.